packages/core/content-manager/server/src/history/services/lifecycles.ts

```diff
@@ -10,6 +10,86 @@ import { FIELDS_TO_IGNORE, HISTORY_VERSION_UID } from '../constants';
 import { CreateHistoryVersion } from '../../../../shared/contracts/history-versions';
 import { createServiceUtils } from './utils';
 
+/**
+ * Filters out actions that should not create a history version.
+ */
+const shouldCreateHistoryVersion = (
+  context: Modules.Documents.Middleware.Context
+): context is Modules.Documents.Middleware.Context & {
+  action: 'create' | 'update' | 'clone' | 'publish' | 'unpublish' | 'discardDraft';
+  contentType: UID.CollectionType;
+} => {
+  // Ignore requests that are not related to the content manager
+  if (!strapi.requestContext.get()?.request.url.startsWith('/content-manager')) {
+    return false;
+  }
+
+  // NOTE: cannot do type narrowing with array includes
+  if (
+    context.action !== 'create' &&
+    context.action !== 'update' &&
+    context.action !== 'clone' &&
+    context.action !== 'publish' &&
+    context.action !== 'unpublish' &&
+    context.action !== 'discardDraft'
+  ) {
+    return false;
+  }
+
+  /**
+   * When a document is published, the draft version of the document is also updated.
+   * It creates confusion for users because they see two history versions each publish action.
+   * To avoid this, we silence the update action during a publish request,
+   * so that they only see the published version of the document in the history.
+   */
+  if (
+    context.action === 'update' &&
+    strapi.requestContext.get()?.request.url.endsWith('/actions/publish')
+  ) {
+    return false;
+  }
```

Comment:
Will this work for bulk publish too?

Reply:
Good point, I forgot about that.

This case doesn't work on bulk publish, but also do not update entries, so it's not necessary either!